platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/Network.java

```diff
@@ -166,52 +121,80 @@ public static void stopPortForwarding() {
         if (ownAddresses == null) {
-            try {
-                ownAddresses = computeOwnAddresses();
-            } catch (SocketException e) {
-                logger.error(EXCEPTION.getMarker(), "", e);
-            }
+            ownAddresses = computeOwnAddresses();
         }
 
-        return addresses;
-    }
-
-    static Collection<InetAddress> getOwnAddresses() throws SocketException {
-        return ownAddresses == null ? ownAddresses = computeOwnAddresses() : ownAddresses;
+        return ownAddresses;
     }
 
     /**
      * Recompute the set of IP addresses for this computer, and their string representations.
      *
      * @return the set of addresses
-     * @throws SocketException
-     * 		for errors while retrieving the addresses
      */
-    private static Collection<InetAddress> computeOwnAddresses() throws SocketException {
+    private static Collection<InetAddress> computeOwnAddresses() {
         final Set<InetAddress> result = new HashSet<>();
-        final List<String> ip4 = new ArrayList<>();
-        final List<String> ip6 = new ArrayList<>();
-
-        for (Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-                interfaces != null && interfaces.hasMoreElements(); ) {
-            for (Enumeration<InetAddress> addresses = interfaces.nextElement().getInetAddresses();
-                    addresses != null && addresses.hasMoreElements(); ) {
-                InetAddress addr = addresses.nextElement();
-                result.add(addr);
-                String ip = addr.getHostAddress();
-                if (ip != null && !ip.equals("127.0.0.1")) {
-                    ((addr instanceof Inet4Address) ? ip4 : ip6).add(ip);
+
+        try {
+            for (final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+                    interfaces.hasMoreElements(); ) {
+                for (final Enumeration<InetAddress> addresses =
+                                interfaces.nextElement().getInetAddresses();
+                        addresses.hasMoreElements(); ) {
+                    InetAddress addr = addresses.nextElement();
+                    result.add(addr);
                 }
             }
+            return result;
+        } catch (SocketException e) {
+            logger.error(EXCEPTION.getMarker(), "Error getting own addresses", e);
+            return Collections.emptySet();
         }
+    }
 
-        Collections.sort(ip4);
-        Collections.sort(ip6);
-        ip4.add("127.0.0.1");
-        ip4.addAll(ip6);
-        addresses = ip4.toArray(new String[0]);
-        return result;
+    /**
+     * Determine whether a given IP address is a private address, as defined by RFC 1918.
+     * This works for both IPv4 and IPv6.
+     *
+     * @param addr the IP address to check
+     *
+     * @return true if addr is a private address
+     */
+    private static boolean isPrivateIP(@NonNull InetAddress addr) {
```

Comment:
Great solution!
But:
I'm missing unit tests for that method. Maybe we create a util class and make the method public in that util class and provide unit tests.

Reply:
Added